src/utils.js

```diff
@@ -71,46 +98,55 @@ function getRelevantESLintConfig(eslintConfig) {
  */
 function getPrettierOptionsFromESLintRules(eslintConfig, prettierOptions) {
   const {rules} = eslintConfig
-  const optionGetters = {
-    printWidth: getPrintWidth,
-    tabWidth: getTabWidth,
-    parser: getParser,
-    singleQuote: getSingleQuote,
-    trailingComma: getTrailingComma,
-    bracketSpacing: getBraketSpacing,
-  }
 
-  return Object.keys(optionGetters).reduce(
-    (options, key) => {
-      const givenOption = prettierOptions[key]
-      const optionIsGiven = prettierOptions[key] !== undefined
-      const getter = optionGetters[key]
-      options[key] = optionIsGiven ? givenOption : getter(rules)
-      return options
-    },
+  return Object.keys(OPTION_GETTERS).reduce(
+    (options, key) => configureOptions(prettierOptions, key, options, rules),
     {},
   )
 }
 
-function getPrintWidth(rules) {
-  return getRuleValue(rules, 'max-len', 80, 'code')
+// If an ESLint rule that prettier can be configured with is enabled create a
+// prettier configuration object that reflects the ESLint rule configuration.
+function configureOptions(prettierOptions, key, options, rules) {
+  const givenOption = prettierOptions[key]
+  const optionIsGiven = givenOption !== undefined
+
+  if (optionIsGiven) {
+    options[key] = givenOption
+  } else {
+    const {ruleValue, ruleValueToPrettierOption} = OPTION_GETTERS[key]
+    const getterRuleValue = ruleValue(rules)
+
+    if (getterRuleValue !== RULE_DISABLED) {
+      const prettierOptionValue = ruleValueToPrettierOption(
+        getterRuleValue,
+        rules,
+      )
+
+      if (prettierOptionValue !== RULE_DISABLED) {
+        options[key] = prettierOptionValue
+      }
+    }
+  }
+
+  return options
 }
 
-function getTabWidth(rules) {
-  const value = getRuleValue(rules, 'indent', 2)
-  // if the value is not a number, default to 2
-  // use-case is 'tab' where prettier doesn't
-  // allow tabs.
-  return typeof value === 'number' ? value : 2
+function getTabWidth(value) {
+  // Fallback to prettier default if `value` is `tab`.
```

Comment:
Could we update the comment to be:

```js
// Fallback to prettier default if `value` is `tab`
// because prettier does not support tabs anyway.
```